scripts/lib/api/generateMdxComponents.ts

```diff
@@ -0,0 +1,361 @@
+  return {
+    id,
+    signature,
+    githubSourceLink,
+  };
+}
+
+function prepareMethodProps(
+  $: CheerioAPI,
+  $child: Cheerio<any>,
+  $dl: Cheerio<any>,
+  priorApiType: string | undefined,
+  githubSourceLink: string,
+  id: string,
+): componentProps {
+  const props = {
+    id,
+    signature: $child.html()!,
+    githubSourceLink,
+  };
+  if (id && !priorApiType) {
+    $dl.siblings("h1").text(getLastPartFromFullIdentifier(id));
+    return props;
+  } else if ($child.attr("id")) {
+    $(`<h3>${getLastPartFromFullIdentifier(id)}</h3>`).insertBefore($dl);
+  }
+
+  return {
+    ...props,
+    name: getLastPartFromFullIdentifier(id),
+  };
+}
+
+function prepareAttributeProps(
+  $: CheerioAPI,
+  $child: Cheerio<any>,
+  $dl: Cheerio<any>,
+  priorApiType: string | undefined,
+  githubSourceLink: string,
+  id: string,
+): componentProps | undefined {
+  if (!priorApiType) {
+    if (id) {
+      $dl.siblings("h1").text(getLastPartFromFullIdentifier(id));
+    }
+
+    const signature = $child.find("em").text()?.replace(/^:\s+/, "");
+    if (signature.trim().length === 0) return;
+    return {
+      id,
+      signature,
+      githubSourceLink,
+    };
+  }
+
+  // Else, the attribute is embedded on the class
+  const text = $child.text();
+
+  // Index of the default value of the attribute
+  let equalIndex = text.indexOf("=");
+  if (equalIndex == -1) {
+    equalIndex = text.length;
+  }
+  // Index of the attribute's type. The type should be
+  // found before the default value
+  let colonIndex = text.slice(0, equalIndex).indexOf(":");
+  if (colonIndex == -1) {
+    colonIndex = text.length;
+  }
+
+  $(`<h3>${getLastPartFromFullIdentifier(id)}</h3>`).insertBefore($dl);
+  // The attributes have the following shape: name [: type] [= value]
+  const name = text.slice(0, Math.min(colonIndex, equalIndex)).trim();
+  const attributeType = text
+    .slice(Math.min(colonIndex + 1, equalIndex), equalIndex)
+    .trim();
+  const attributeValue = text.slice(equalIndex, text.length).trim();
+
+  return {
+    id,
+    name,
+    attributeType,
+    attributeValue,
+  };
+}
+
+function prepareFunctionOrExceptionProps(
+  $: CheerioAPI,
+  $child: Cheerio<any>,
+  $dl: Cheerio<any>,
+  id: string,
+  githubSourceLink: string,
+): componentProps {
+  const props = {
+    id,
+    signature: $child.html()!,
+    githubSourceLink,
+  };
+
+  const pageHeading = $dl.siblings("h1").text();
+  if (id.endsWith(pageHeading) && pageHeading != "") {
+    // Page is already dedicated to apiType; no heading needed
+    return props;
+  }
+  $(`<h3>${getLastPartFromFullIdentifier(id)}</h3>`).insertBefore($dl);
+
+  return {
+    ...props,
+    name: id.split(".").slice(-1)[0],
+  };
+}
+
+// ------------------------------------------------------------------
+// Generate MDX components
+// ------------------------------------------------------------------
+
+export async function generateMdxComponent(
```

Comment:
Single function to generate the component open tag. If any prop is empty will be removed by the unified plugin on `htmlToMd.ts`

Reply:
Nice. This would be good to reflect in the code:

1. Consider calling the function `createOpeningTag`
2. It'd be good to have a comment above the JSX explaining that about removing empty props.